src/benchmark/operators/table_scan_sorted_benchmark.cpp

```diff
@@ -0,0 +1,277 @@
+      }
+    }
+  }
+
+  if (encoding_type != EncodingType::Unencoded) {
+    ChunkEncoder::encode_all_chunks(table, SegmentEncodingSpec(encoding_type));
+  }
+
+  if (order_by.has_value()) {
+    for (auto& chunk : table->chunks()) {
+      chunk->get_segment(ColumnID(0))->set_sort_order(order_by.value());
+    }
+  }
+
+  table_wrapper = std::make_shared<TableWrapper>(std::move(table));
+  table_wrapper->execute();
+
+  return table_wrapper;
+}
+
+std::shared_ptr<TableWrapper> create_string_table(const int table_size, const int string_length,
+                                                  const EncodingType encoding_type = EncodingType::Unencoded,
+                                                  const std::optional<OrderByMode> order_by = std::nullopt) {
+  const auto table_column_definitions = create_column_definitions(DataType::String);
+  std::shared_ptr<Table> table;
+  std::shared_ptr<TableWrapper> table_wrapper;
+
+  table = std::make_shared<Table>(*table_column_definitions, TableType::Data);
+
+  if (!order_by.has_value()) {
+    auto values = std::vector<std::string>(table_size);
+    for (int i = 0; i < table_size; i++) {
+      auto str = std::to_string(i);
+      str = std::string(string_length - str.length(), '0').append(str);
+      values[i] = str;
+    }
+    std::random_shuffle(values.begin(), values.end());
+    for (const auto& str : values) {
+      table->append({str});
+    }
+  } else if (order_by.value() == OrderByMode::Ascending || order_by.value() == OrderByMode::AscendingNullsLast) {
+    for (int i = 0; i < table_size; i++) {
+      auto str = std::to_string(i);
+      str = std::string(string_length - str.length(), '0').append(str);
+      table->append({str});
+    }
+  } else {
+    for (int i = 0; i < table_size; i++) {
+      auto str = std::to_string(table_size - i - 1);
+      str = std::string(string_length - str.length(), '0').append(str);
+      table->append({str});
+    }
+  }
+
+  if (encoding_type != EncodingType::Unencoded) {
+    ChunkEncoder::encode_all_chunks(table, SegmentEncodingSpec(encoding_type));
+  }
+
+  if (order_by.has_value()) {
+    for (auto& chunk : table->chunks()) {
+      chunk->get_segment(ColumnID(0))->set_sort_order(order_by.value());
+    }
+  }
+
+  table_wrapper = std::make_shared<TableWrapper>(std::move(table));
+  table_wrapper->execute();
+
+  return table_wrapper;
+}
+
+void BM_TableScanSorted(
+    benchmark::State& state, const int table_size, const double selectivity,
+    const PredicateCondition predicate_condition, const EncodingType encoding_type,
+    const std::optional<OrderByMode> order_by,
+    std::function<std::shared_ptr<TableWrapper>(const int, const EncodingType, const OrderByMode)> table_creator) {
+  _clear_cache();
+
+  AllTypeVariant search_value;
+  switch (predicate_condition) {
+    case PredicateCondition::LessThanEquals:
+    case PredicateCondition::LessThan:
+      search_value = static_cast<int>(table_size * selectivity);
```

Comment:
Without having looked any further, I don't understand what exactly you are doing here. I can only guess.

Reply:
The benchmarks all run with different selectivities (ratio of values in the output to values in the input). At this point the search value is selected in a way that our results corresponds to the chosen selectivity.

Reply:
You could add that sentence as a comment.